### PR TITLE
[Add tests] To hotfix regarding the saving of `None` metadata

### PR DESCRIPTION
**Pull request by dbogunowicz**
Tests related to https://github.com/neuralmagic/sparseml/pull/660.

Asserting that:

1. While Rahul's fix corrects the behavior of function `metadata_to_string_lines` in `optim/manager.py` for an unstaged recipe, I am making sure that the logic is foolproof for staged recipes.
2. Once we take a recipe without metadata (staged or non-staged) and save it using `manager.save()`, the resulting YAML file can be once again correctly digestible when reused. I was mostly concerned about what happens if we consume a recipe with `__metadata__:   <empty> `. Fortunately, it is being parsed correctly.

- [x] use str(manager) instead of saving recipes across all metadata tests

### CODE REVIEW

File: tests/sparseml/pytorch/optim/test_manager.py [hunk 1]
```diff
@@ -468,6 +468,68 @@ def test_lifecycle_manager_staged(
...
+@pytest.mark.parametrize(
+    "recipe",
+    [
+        STANDARD_RECIPE_1,  # passing standard recipe without metadata
+        TWO_STAGES_RECIPE,  # passing staged recipe without metadata
+    ],
+)
+def test_lifecycle_manager_staged_no_metadata(recipe):
+    temp_dir = tempfile.mkdtemp()
+    recipe_path = os.path.join(temp_dir, "recipy.yaml")
```
Comment (rahul-tuli) [inline tests/sparseml/pytorch/optim/test_manager.py:518]: nit: Maybe this is a typo
```suggestion
    recipe_path = os.path.join(temp_dir, "recipe.yaml")
```

Comment (dbogunowicz) [reply tests/sparseml/pytorch/optim/test_manager.py:518]: Yeah, I keep making this same typo over and over 🥇

**Review by bfineran — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]
